hoomd/hpmc/tune/move_size.py

```diff
@@ -0,0 +1,333 @@
+from hoomd.custom import _InternalAction
+from hoomd.parameterdicts import ParameterDict
+from hoomd.typeconverter import OnlyFrom, OnlyType, OnlyIf, to_type_converter
+from hoomd.tune import _InternalCustomTuner
+from hoomd.tune.attr_tuner import (
+    _TuneDefinition, SolverStep, ScaleSolver, SecantSolver)
+from hoomd.hpmc.integrate import _HPMCIntegrator
+
+
+class _MoveSizeTuneDefinition(_TuneDefinition):
+    """Encapsulates getting the acceptance rate and getting/setting move size.
+
+    This class should only be used for the _InternalMoveSize class to tune HPMC
+    move sizes. For this class 'x' is the move size and 'y' is the acceptance
+    rate.
+    """
+    _attr_acceptance = {
+        'a': 'rotate_moves',
+        'd': 'translate_moves'
+        }
+
+    def __init__(self, attr, type, target, domain=None):
+        if attr not in self._attr_acceptance:
+            raise ValueError("Only {} are allowed as tunable "
+                             "attributes.".format(self._available_attrs))
+        self.attr = attr
+        self.type = type
+        self.integrator = None
+        self.previous_ratio = None
+        self.previous_total = None
+        super().__init__(target, domain)
+
+    def _get_y(self):
+        current_ratio = getattr(self.integrator,
+                                self._attr_acceptance[self.attr])
+        current_total = sum(current_ratio)
+        # We return the target when no moves are recorded since we don't want
+        # the move size to be updated. This is "hackish", but there is no right
+        # answer here since there is no concept of an acceptance_rate with no
+        # trial moves. We could error, but I think this make the class easier to
+        # use for users.
+        if current_total == 0:
+            return self._target
+        # If no more trial moves have been recorded return previous
+        # acceptance_rate. In general, this conditional should not be true, if
+        # it is true different solvers may error, but this is the most natural
+        # solution, I could think of.
+        if (self.previous_total is not None
+                and self.previous_total == current_total):
+            return self.previous_ratio[0] / self.previous_total
+
+        if self.previous_ratio is None or self.previous_total > current_total:
+            acceptance_rate = current_ratio[0] / current_total
```

Comment:
We can tune here because the move counts are non-zero. On the first call, we can tune as long as there are moves that have been performed. Likewise, when the counters are lower than before, they have been reset, but the existing count does give us information about the acceptance rate. In both cases, we should tune I believe since we have information to do so.

The only case we don't tune is if there are no moves to get an acceptance rate for. If I am not understanding your point let me know.

Reply:
Let me clarify with an example. Take this case:
```
sim.operations.tuners.append(MoveSizeTuner(..., trigger=And(After(1e8), Periodic(100)))
run(1e9)
```
Since the reading uses `current_total`, it will be the averaged value over the first 1e8 steps. The second reading will be computed from only the 100 steps after making the change. This is the inconsistency I am referring to. Depending on what happened before tuning started (i.e. compression), that first average reading may not actually reflect the current acceptance rate of the system.

Reply:
I see your point. We shouldn't tune on the first pass.  I think this makes a more compelling argument for allowing y to be `None` as well as a way of denoting to skip this for now.